openedx/features/badging/models.py

```diff
@@ -120,13 +130,16 @@ def assign_badge(cls, user_id, badge_id, community_id):
                 raise Exception(error)
 
             all_team_members = CourseTeamMembership.objects.filter(team_id=team_group_chat['team_id'])
+            assigned = False
             for member in all_team_members:
-                UserBadge.objects.get_or_create(
+                user_badge, assigned = UserBadge.objects.get_or_create(
                     user_id=member.user_id,
                     badge_id=badge_id,
                     course_id=course_id,
                     community_id=community_id
                 )
+
+            return assigned
```

Comment:
This will only return the value of assigned for the last member for example if 5 users were successfully created and the 6th failed, the view sends back a conflict. Maybe we should either use any or maybe we can check if the current user was assigned badge or not.